Newer gnome-keyring may raise CancelledError

In such case let's continue without the keyring instead of crashing.


  Patch attached to upstream trac ticket:
  http://trac.gajim.org/ticket/5633
  https://bugzilla.redhat.com/show_bug.cgi?id=556374
    -- Michal

Index: gajim-0.13.3/src/common/passwords.py
===================================================================
--- gajim-0.13.3.orig/src/common/passwords.py
+++ gajim-0.13.3/src/common/passwords.py
@@ -197,7 +197,7 @@ def get_storage():
 		if USER_USES_GNOMEKEYRING:
 			try:
 				storage = GnomePasswordStorage()
-			except (gnomekeyring.NoKeyringDaemonError, gnomekeyring.DeniedError):
+			except (gnomekeyring.NoKeyringDaemonError, gnomekeyring.DeniedError, gnomekeyring.CancelledError):
 				storage = None
 		if storage is None:
 			if gajim.config.get('use_kwalletcli'):
